src/main/java/org/infernalstudios/shieldexp/network/SyncShields.java

```diff
@@ -40,9 +38,8 @@ public void handle(Supplier<NetworkEvent.Context> ctx){
 
     @OnlyIn(Dist.CLIENT)
     private void handle() {
-        Player player = Minecraft.getInstance().player;
-        if (player != null){
-            ShieldDataLoader.parse(shield, data.getAsJsonObject());
-        }
+        // trust that the client can handle this even if the player isn't ready
+        // this data isn't stored to the local player so even if the player is null, it's fine
+        ShieldDataLoader.parse(shield, data.getAsJsonObject());
```

Comment:
Note, not applicable to this PR: This should be reworked in the future, as using JSON to store and send these values is an unnecessary abstraction, we can just store straight doubles in the packet. The `data` object is fixed-size.

Reply:
Yes, I agree. The goal of the PR is to fixup the current system, not to reimplement it, so this solution will work for now. Honestly the correct way to do this is with some sort of custom data registry that can be synced kind of like many others in the game.